### PR TITLE
Get disc status

### LINKED ISSUE #20: Agent: get good/bad status from  system

### PR DESCRIPTION
**Pull request by wkucjas**
It should fix issue #20

### CODE REVIEW

File: src/agent/CMakeLists.txt [hunk 1]
```diff
@@ -3,13 +3,13 @@ find_package(Qt5 5.15 REQUIRED COMPONENTS Core Network)
...
-				HealthChecker.h
-				HealthChecker.cpp
+				
+				
 				ISmartReader.h
 				SmartReader.h
                 Server.h
                 Server.cpp
-                )
+                 "windows/CMDCommunication.h" "windows/CMDCommunication.cpp")
```
Comment (Kicer86) [inline src/agent/CMakeLists.txt]: please follow current convention

File: src/agent/ISmartReader.h [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+
 #include <string>
+#include "common/GeneralHealth.h"
 
 class ISmartReader
 {
 public:
 	virtual ~ISmartReader() = default;
 	virtual std::string ReadSMARTData() = 0;
+	virtual GeneralHealth ReadDisksStatus() = 0;
```
Comment (Kicer86) [inline src/agent/ISmartReader.h:11]: I believe there should be another tool for calculating health. Now SmartReader will have two responsibilities 
which makes it harder to reuse.

File: src/agent/Server.cpp [hunk 3]
```diff
@@ -62,6 +63,7 @@ void Server::SendData()
 
 void Server::CollectInfoAboutDiscs()
 {
-    m_health.SetStatus( GeneralHealth::GOOD);
+    SmartReader reader;
+    m_health = reader.ReadDisksStatus();
```
Comment (Kicer86) [inline src/agent/Server.cpp:67]: Not an issue yet, but Server should ask for status via injected interface so we can have loose coupling here.

File: src/agent/windows/CMDCommunication.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        m_generalHealth.SetStatus(GeneralHealth::Health::BAD);
+    }
+
+    return m_generalHealth;
+}
+
+std::string CMDCommunication::ExecuteDiscStatusCommand() const
+{
+    std::string command = "wmic diskdrive get status";
+    std::string fileName = "dh.txt";
```
Comment (Kicer86) [inline src/agent/windows/CMDCommunication.cpp]: There is a nice Qt class for creating custom dirs and files: QTemporaryDir and QTemporaryFile. Consider them.

Comment (Kicer86) [reply src/agent/windows/CMDCommunication.cpp]: Also, w QProcess you can read process' output directly without a file

**Review by Kicer86 — CHANGES_REQUESTED** [review]

File: src/agent/ISmartReader.h (hunk 2 above)
Comment (wkucjas) [reply src/agent/ISmartReader.h:11]: This will be changed in future task.

File: src/agent/Server.cpp (hunk 3 above)
Comment (wkucjas) [reply src/agent/Server.cpp:67]: This will be changed in future task.

File: src/agent/windows/CMDCommunication.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+std::string CMDCommunication::ExecuteDiscStatusCommand() const
+{
+    QProcess proc;
+    proc.start("wmic diskdrive get status");
+    proc.waitForFinished();
+    QString output = proc.readAllStandardOutput();
+
+    std::string ret = output.toStdString();
+    std::string::iterator retEnd = ret.end();
+    for (auto i = ret.begin(); i < retEnd; ++i)
```
Comment (Kicer86) [inline src/agent/windows/CMDCommunication.cpp]: would QString::trimmed()  be enought?

**Review by Kicer86 — CHANGES_REQUESTED** [review]
Check if code can be simplified

**Review by Kicer86 — APPROVED** [review]

**Pull request merged**